tests/test_measurements.py

```diff
@@ -658,191 +679,164 @@ def circuit():
 
         circuit()
 
-    def test_binned_counts_for_operator(self, tol):
-        """Test that the shape is correct with the shot vector."""
-        shot_vec = (10, 10)
-        dev = qml.device("default.qubit", wires=3, shots=shot_vec)
-
-        @qml.qnode(dev)
-        def circuit():
-            qml.Hadamard(wires=0)
-            qml.PauliZ(0)
-            return qml.sample(qml.PauliZ(0), counts=True)
-
-        binned_samples = circuit()
-
-        assert isinstance(binned_samples, np.ndarray)
-        assert isinstance(binned_samples[0], dict)
-        assert len(binned_samples) == len(shot_vec)
-        assert sum(sum(v for v in bin.values()) for bin in binned_samples) == sum(shot_vec)
-
-    def test_binned_counts_for_state_vector(self, tol):
-        """Test that the shape is correct with the shot vector and no observable too."""
-        shot_vec = (10, 10)
-        dev = qml.device("default.qubit", wires=3, shots=shot_vec)
-
-        @qml.qnode(dev)
-        def circuit():
-            qml.Hadamard(wires=0)
-            qml.PauliZ(0)
-            return qml.sample(None, counts=True)
-
-        binned_samples = circuit()
-
-        assert isinstance(binned_samples, np.ndarray)
-        assert isinstance(binned_samples[0], dict)
-        assert len(binned_samples) == len(shot_vec)
-        assert sum(sum(v for v in bin.values()) for bin in binned_samples) == sum(shot_vec)
-
-    def test_counts_jax(self, tol):
-        """Check jax interface with computational basis state counts"""
-        n_shots = 10
-        dev = qml.device("default.qubit", wires=3, shots=n_shots)
-
-        @qml.qnode(dev, inteface="jax")
-        def circuit():
-            return qml.sample(counts=True)
-
-        res = circuit()
-        assert res == {"000": n_shots}
-
-    def test_counts_operator_jax(self, tol):
-        """Check jax interface with observable measurement counts"""
-        n_shots = 10
-        dev = qml.device("default.qubit", wires=3, shots=n_shots)
-
-        @qml.qnode(dev, inteface="jax")
-        def circuit():
-            return qml.sample(qml.PauliZ(0), counts=True)
-
-        res = circuit()
-        assert res == {1: n_shots}
-
-    def test_counts_tf(self, tol):
-        """Check TensorFlow interface with computational basis state counts"""
+    @pytest.mark.all_interfaces
+    @pytest.mark.parametrize("wires, basis_state", [(None, "010"), ([2, 1], "01")])
+    @pytest.mark.parametrize("interface", ["autograd", "jax", "tensorflow", "torch"])
+    def test_counts_no_op_finite_shots(self, interface, wires, basis_state, tol):
+        """Check all interfaces with computational basis state counts and
+        finite shot"""
         n_shots = 10
         dev = qml.device("default.qubit", wires=3, shots=n_shots)
 
-        @qml.qnode(dev, inteface="tf")
+        @qml.qnode(dev, interface=interface)
```

Comment:
Originally a typo (having `inteface` instead of `interface`) made a major difference in testing as none of the interfaces were being tested in practice.

Reply:
If this was an issue, I wonder if we need better keyword validation for qnodes? The number of keywords for qnodes is extensive, and it's a definite code smell.